### PR TITLE
TEST-#7064: Explicitly check for exceptions in `test_groupby.py`

### LINKED ISSUE #7064: Explicitly check for exceptions in `test_groupby.py`

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #7064  
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/test/test_groupby.py [hunk 1]
```diff
@@ -461,7 +477,7 @@ def test_aggregate_alias():
 @pytest.mark.parametrize(
     "col1_category", [True, False], ids=lambda v: f"col1_category={v}"
 )
-def test_simple_row_groupby(by, as_index, col1_category):
+def test_simple_row_groupby(by, as_index, col1_category, request):
```
Comment (YarShev) [inline modin/pandas/test/test_groupby.py:480]: Why is request added? Do we use it somewhere?

File: modin/pandas/test/test_groupby.py [hunk 2]
```diff
@@ -3048,7 +3187,7 @@ def test_reshuffling_groupby_on_strings(modify_config):
     pd_grp = pandas_df.groupby("col1")
 
     eval_general(md_grp, pd_grp, lambda grp: grp.mean())
-    eval_general(md_grp, pd_grp, lambda grp: grp.nth())
+    eval_general(md_grp, pd_grp, lambda grp: grp.nth(2))
```
Comment (YarShev) [inline modin/pandas/test/test_groupby.py:3190]: Why changed?

File: modin/pandas/test/test_groupby.py [hunk 3]
```diff
@@ -2837,8 +2966,14 @@ def eval_rolling(md_window, pd_window):
     eval_general(md_window, pd_window, lambda window: window.std())
     eval_general(md_window, pd_window, lambda window: window.min())
     eval_general(md_window, pd_window, lambda window: window.max())
-    eval_general(md_window, pd_window, lambda window: window.corr())
-    eval_general(md_window, pd_window, lambda window: window.cov())
+    # FIXME: identify error
```
Comment (YarShev) [inline modin/pandas/test/test_groupby.py:2969]: Do you plan to identify the error in this PR?

File: modin/pandas/test/test_groupby.py (hunk 2 above)
Comment (anmyachev) [reply modin/pandas/test/test_groupby.py:3190]: Required parameter: https://pandas.pydata.org/docs/reference/api/pandas.core.groupby.DataFrameGroupBy.nth.html#pandas.core.groupby.DataFrameGroupBy.nth

File: modin/pandas/test/test_groupby.py (hunk 1 above)
Comment (anmyachev) [reply modin/pandas/test/test_groupby.py:480]: Removed

File: modin/pandas/test/test_groupby.py (hunk 3 above)
Comment (anmyachev) [reply modin/pandas/test/test_groupby.py]: yep, done

**Comment by anmyachev** [comment]
HDK tests need to be restarted

File: modin/pandas/test/test_groupby.py [hunk 4]
```diff
@@ -510,24 +526,85 @@ def maybe_get_columns(df, by):
                 pandas_groupby,
                 lambda df: df.nth(0).sort_values("col1").reset_index(drop=True),
             )
+
+    raising_exceptions = None
+    if col1_category:
```
Comment (YarShev) [inline modin/pandas/test/test_groupby.py:531]: In all other cases there are no exceptions, right? I mean don't we hide the errors by setting raising_exceptions=None?

Comment (anmyachev) [reply modin/pandas/test/test_groupby.py:531]: Should be the same answer like in https://github.com/modin-project/modin/pull/7067#discussion_r1522033510

**Review by YarShev — APPROVED** [review]

**Pull request merged**